update

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -ex
+
+BRANCH=${1:-master}
+
+git clean -f -d
+GENERATED_PATHS=$(ls -1)
+
+# checkout source branch
+git checkout "$BRANCH" && git submodule update --init
+COPY_PATHS=$(ls -1)
+tag=$(git rev-parse HEAD)
+
+# copy the branch content into source/
+mkdir source
+for i in $COPY_PATHS; do
+    cp -r "$i" source
+done
+VERSIONS=$(sed -n 's/^VERSIONS[[:space:]]*=//p' source/Makefile)
+
+git checkout generated
+# Sometimes git does not remove common/ on branch change
+[ -d common ] && rm -rf common
```

Comment:
This wouldn't be needed if we started from scratch, by dangling commit, has this been considered?  Plus, it would be somewhat desirable in future to move the update script to common scripts.

Reply:
I did not see much added value to starting from scratch other than to work around the issue. But you have a point that it would make sense to add update into common scripts instead so I will do that now (since at least python will need it in near future).